packages/lesswrong/server/scripts/convertImagesToCloudinary.ts

```diff
@@ -40,88 +41,122 @@ async function moveImageToCloudinary(oldUrl: string, originDocumentId: string):
   return result.url;
 }
 
-// Images on domains not in this list will be mirrored on Cloudinary and have
-// their lines updated. (If you run the script. This doesn't (yet) auto-apply
-// to all posts.)
-const imageUrlWhitelist = [
-  "cloudinary.com",
-  "res.cloudinary.com",
-  "www.lesswrong.com",
-  "www.alignmentforum.org",
-];
+/**
+ * Images on domains not in this list will be mirrored on Cloudinary and have
+ * their src updated.
+ */
+function getImageUrlWhitelist() {
+  const localUploadUrl = ckEditorUploadUrlOverrideSetting.get() || ckEditorUploadUrlSetting.get()
+  return [
+    "cloudinary.com",
+    "res.cloudinary.com",
+    "www.lesswrong.com",
+    "www.alignmentforum.org",
+  ].concat(localUploadUrl ? [new URL(localUploadUrl).host] : []);
+}
 
-function urlNeedsMirroring(url: string) {
+function urlNeedsMirroring(url: string, filterFn: (url: string) => boolean) {
   const parsedUrl = new URL(url);
-  if (imageUrlWhitelist.indexOf(parsedUrl.hostname) !== -1)
+  if (getImageUrlWhitelist().indexOf(parsedUrl.hostname) !== -1) {
     return false;
-  return true;
+  }
+  return filterFn(url);
 }
 
-async function convertImagesInHTML(html: string, originDocumentId: string): Promise<string> {
+async function convertImagesInHTML(html: string, originDocumentId: string, urlFilterFn: (url: string) => boolean = () => true): Promise<{count: number, html: string}> {
   const parsedHtml = cheerio.load(html);
-  const imgTags = parsedHtml("img");
-  
+  const imgTags = parsedHtml("img").toArray();
+
+  // Upload all the images to Cloudinary (slow)
+  const mirrorUrls = await Promise.all(imgTags.map(async tag => {
+    const src = cheerio(tag).attr("src")
+    if (!src || !(src && urlNeedsMirroring(src, urlFilterFn))) return null
+
+    const newUrl = await moveImageToCloudinary(src, originDocumentId)
```

Comment:
Maybe? Then the lambda doesn't need to be async anymore too.
```suggestion
    return moveImageToCloudinary(src, originDocumentId)
```